backends-velox/src/main/scala/org/apache/gluten/execution/HashAggregateExecTransformer.scala

```diff
@@ -356,7 +356,7 @@ abstract class HashAggregateExecTransformer(
           // The process of handling the inconsistency in column types and order between
           // Spark and Velox is exactly the opposite of applyExtractStruct.
           aggregateExpression.mode match {
-            case PartialMerge | Final =>
+            case PartialMerge | Final | Complete =>
```

Comment:
Have you tested the complete Agg? This may cause failure on rewrite. I have tested the #6198 , and it has run through all TPCDS queries with `Partial Agg Pushed down`, which would lead many Complete Agg Nodes.

Reply:
I added [unit tests](https://github.com/apache/incubator-gluten/pull/7752/files#diff-742ce9027f0d774d324125a130d9993bad09d4ea5eb6d35d791b39bb57138c77R1260) to check data consistency, that the operator is executed as HashAggregateExecBaseTransformer, that substrait plan is in the COMPLETE mode, and that there are existing TPC-H test execution plans and unit tests. cc @zhouyuan 
![image](https://github.com/user-attachments/assets/e324860f-768f-490c-a92b-a34183170bf9)


In velox backend, the mode of the aggregate function depends on the core::AggregationNode::Step, I understand the right aggregation patterns are pushed down on it. And I'm sorry that I don't quite understand the problem `which would lead many Complete Agg Nodes`. Could you explain it in more detail?

Reply:
[Partial Agg Pushed down](https://github.com/apache/spark/pull/36552) is an unmerged Pr in Apache/Spark, we have used it in our environment and found it add many COMPLETE Mode Agg to SQL Plans. I used #6198 to support COMPLETE Mode for velox before. If ok, I can do double check for your pr to see if there is some prolem with it.

Reply:
It seems to be push down in the optimizer phase, where partial and final aggs are not connected aggregates, and the merge agg theoretically won't handle this pattern. I'm guessing you mean the modification in the HashAggregateExecTransformer that supports the COMPLETE part, okay, you can help me check that, thank you.

Reply:
Current code works with our environment, thanks.